google-cloud-storage/src/test/java/com/google/cloud/storage/BlobTest.java

```diff
@@ -662,4 +666,38 @@ public Long answer() throws Throwable {
     byte actual[] = Files.readAllBytes(file.toPath());
     assertArrayEquals(expected, actual);
   }
+
+  @Test
+  public void testUploadNonExistentFile() throws Exception {
+    initializeExpectedBlob(1);
+    expect(storage.getOptions()).andReturn(mockOptions);
+    replay(storage);
+    blob = new Blob(storage, new BlobInfo.BuilderImpl(BLOB_INFO));
+    String fileName = "non_existing_file.txt";
+    try {
+      blob.uploadFrom(Paths.get(fileName));
+    } catch (IllegalArgumentException e) {
+      assertEquals("file should exist " + fileName, e.getMessage());
+    }
+  }
+
+  @Test
+  public void testUpload() throws Exception {
```

Comment:
here test method name shoud be `testUploadFrom()`

Reply:
I named in this way to be consistent with existing tests:
`testDownload()`
`testDownloadWithRetries()`

Reply:
I renamed testDownload => testDownloadTo and testUpload => testUploadFrom